SPECS/kernel-uvm/config

```diff
@@ -1614,7 +1615,38 @@ CONFIG_SCSI_VIRTIO=y
 # end of SCSI device support
 
 # CONFIG_ATA is not set
-# CONFIG_MD is not set
+CONFIG_MD=y
+# CONFIG_BLK_DEV_MD is not set
+# CONFIG_BCACHE is not set
+CONFIG_BLK_DEV_DM_BUILTIN=y
+CONFIG_BLK_DEV_DM=y
+# CONFIG_DM_DEBUG is not set
+CONFIG_DM_BUFIO=y
+# CONFIG_DM_DEBUG_BLOCK_MANAGER_LOCKING is not set
+# CONFIG_DM_UNSTRIPED is not set
+# CONFIG_DM_CRYPT is not set
+# CONFIG_DM_SNAPSHOT is not set
+# CONFIG_DM_THIN_PROVISIONING is not set
+# CONFIG_DM_CACHE is not set
+# CONFIG_DM_WRITECACHE is not set
+# CONFIG_DM_EBS is not set
+# CONFIG_DM_ERA is not set
+# CONFIG_DM_CLONE is not set
+# CONFIG_DM_MIRROR is not set
+# CONFIG_DM_RAID is not set
+# CONFIG_DM_ZERO is not set
+# CONFIG_DM_MULTIPATH is not set
+# CONFIG_DM_DELAY is not set
+# CONFIG_DM_DUST is not set
+# CONFIG_DM_INIT is not set
+# CONFIG_DM_UEVENT is not set
+# CONFIG_DM_FLAKEY is not set
+CONFIG_DM_VERITY=y
+# CONFIG_DM_VERITY_VERIFY_ROOTHASH_SIG is not set
+# CONFIG_DM_VERITY_FEC is not set
```

Comment:
The Mariner build tools do support setting FEC, but not critical to enable.

Reply:
Alright, I'll leave this unset for now until it's needed

Reply:
I was just comparing related standard Mariner kernel options in SPECS/kernel/config with this kernel. There is a few other divergences as well, e.g. FLAKEY. While it's generally good to align sources as much as possible, I think it's fine to diverge in this case as we seem to enable fewer options and as this is the UVM kernel its size/TCB will have to be reduced as much as possible going forward.

Do you see any advantages for this case for enabling FEC here? The UVM kernel will run as a nested VM on Azure.

Reply:
Sorry for the delta, lost track of the replies.

FEC (forward error correction) is nice in that random bit flips won't kill your verity disk. This is MUCH more important for embedded/IoT/Edge devices rather than cloud since the storage you are running on will likely have error correction built in already (technically sd cards etc are meant to as well, but...).

I'm not 100% sure what the behaviour of the verity tools are if you ask them to use FEC and the kernel doesn't support it, but it should be sufficient to just never pass the FEC arguments at all.